plugins/com.google.cloud.tools.eclipse.appengine.deploy/src/com/google/cloud/tools/eclipse/appengine/deploy/CleanupOldDeploysJob.java

```diff
@@ -0,0 +1,82 @@
+package com.google.cloud.tools.eclipse.appengine.deploy;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.Iterator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.core.runtime.jobs.Job;
+
+import com.google.cloud.tools.eclipse.util.io.DeleteAllVisitor;
+import com.google.cloud.tools.eclipse.util.status.StatusUtil;
+
+public class CleanupOldDeploysJob extends Job {
+
+  private static String NAME = Messages.getString("cleanup.deploy.job.name"); //$NON-NLS-1$
+  private static int OLD_DIRECTORIES_TO_KEEP = 2;
+  private IPath parentTempDir;
+
+  public CleanupOldDeploysJob(IPath parentTempDir) {
+    super(NAME);
+    this.parentTempDir = parentTempDir;
+  }
+
+  @Override
+  protected IStatus run(IProgressMonitor monitor) {
+    try {
+      SortedSet<File> directories = collectDirectories();
+      removeRecentDirectories(directories);
+      deleteDirectories(directories);
+      return Status.OK_STATUS;
+    } catch (IOException e) {
+      return StatusUtil.error(this, Messages.getString("cleanup.deploy.job.error"), e); //$NON-NLS-1$
+    }
+  }
+
+  private SortedSet<File> collectDirectories() throws IOException {
```

Comment:
Make this a TreeSet, not a SortedSet; and then you can just call pollLast twice to remove the two most recent entries rather than iterating through it again in removeRecentDirectories. (or maybe pollFirst depending on which direction you're sorting)